freetextresponse/freetextresponse.py

```diff
@@ -0,0 +1,437 @@
+        self.display_correctness = data['display_correctness']
+        if data['min_word_count'] and data['max_word_count']:
+            if (
+                    int(float(data['min_word_count'])) >= 0 and
+                    (
+                        int(float(data['min_word_count'])) <=
+                        int(float(data['max_word_count']))
+                    )
+            ):
+                self.min_word_count = int(float(data['min_word_count']))
+            if (
+                    int(float(data['max_word_count'])) >= 0 and
+                    (
+                        int(float(data['max_word_count'])) >=
+                        int(float(data['min_word_count']))
+                    )
+            ):
+                self.max_word_count = int(float(data['max_word_count']))
+        self.fullcredit_keyphrases = data['fullcredit_keyphrases']
+        self.halfcredit_keyphrases = data['halfcredit_keyphrases']
+        return {
+            'display_name': self.display_name,
+            'prompt': self.prompt,
+            'weight': self.weight,
+            'max_attempts': self.max_attempts,
+            'display_correctness': self.display_correctness,
+            'min_word_count': self.min_word_count,
+            'max_word_count': self.max_word_count,
+            'fullcredit_keyphrases': self.fullcredit_keyphrases,
+            'halfcredit_keyphrases': self.halfcredit_keyphrases,
+        }
+
+    @classmethod
+    def get_resource_string(cls, path):
+        '''
+        Retrieve string contents for the file path
+        '''
+        path = os.path.join('public', path)
+        resource_string = pkg_resources.resource_string(__name__, path)
+        return resource_string.decode('utf8')
+
+    def get_resource_url(self, path):
+        '''
+        Retrieve a public URL for the file path
+        '''
+        path = os.path.join('public', path)
+        resource_url = self.runtime.local_resource_url(self, path)
+        return resource_url
+
+    def build_fragment(
+            self,
+            html_source=None,
+            paths_css=[],
+            paths_js=[],
+            urls_css=[],
+            urls_js=[],
+            fragment_js=None,
+    ):
+        #  pylint: disable=dangerous-default-value, too-many-arguments
+        '''
+        Assemble the HTML, JS, and CSS for an XBlock fragment
+        '''
+        fragment = Fragment(html_source)
+        for url in urls_css:
+            fragment.add_css_url(url)
+        for path in paths_css:
+            url = self.get_resource_url(path)
+            fragment.add_css_url(url)
+        for url in urls_js:
+            fragment.add_javascript_url(url)
+        for path in paths_js:
+            url = self.get_resource_url(path)
+            fragment.add_javascript_url(url)
+        if fragment_js:
+            fragment.initialize_js(fragment_js)
+        return fragment
+
+    def get_indicator_visiblity_class(self):
```

Comment:
Are these methods intended to be private/local? If so, can we indicate so by prefixing the names with an underscore? `def _get_indicator_visibility_class(self):`